agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -258,6 +272,79 @@ func (state *DockerTaskEngineState) RemoveENIAttachment(mac string) {
 	}
 }
 
+// AllEBSAttachments returns all the ebs volumes managed by ecs on the instance
+func (state *DockerTaskEngineState) AllEBSAttachments() []*apiebs.ResourceAttachment {
+	state.lock.RLock()
+	defer state.lock.RUnlock()
+
+	return state.allEBSAttachmentsUnsafe()
+}
+
+func (state *DockerTaskEngineState) allEBSAttachmentsUnsafe() []*apiebs.ResourceAttachment {
+	var allEBSAttachments []*apiebs.ResourceAttachment
+	for _, v := range state.ebsAttachments {
+		allEBSAttachments = append(allEBSAttachments, v)
+	}
+	return allEBSAttachments
+}
+
+// AllPendingEBSAttachments returns all the ebs volumes managed by ecs on the instance that haven't sent a state change yet
+func (state *DockerTaskEngineState) AllPendingEBSAttachments() []*apiebs.ResourceAttachment {
+	state.lock.RLock()
+	defer state.lock.RUnlock()
+
+	return state.allPendingEBSAttachmentsUnsafe()
+}
+
+func (state *DockerTaskEngineState) allPendingEBSAttachmentsUnsafe() []*apiebs.ResourceAttachment {
+	var allEBSAttachments []*apiebs.ResourceAttachment
+	for _, v := range state.ebsAttachments {
+		if !v.IsSent() {
+			allEBSAttachments = append(allEBSAttachments, v)
+		}
+	}
+	return allEBSAttachments
+}
+
+// AddEBSAttachment adds the ebs volume to state
+func (state *DockerTaskEngineState) AddEBSAttachment(ebsAttachment *apiebs.ResourceAttachment) {
+	if ebsAttachment == nil {
+		seelog.Debug("Cannot add empty ebs attachment information")
+		return
+	}
+	state.lock.Lock()
+	defer state.lock.Unlock()
+	if _, ok := state.ebsAttachments[ebsAttachment.AttachmentProperties[apiebs.VolumeIdName]]; !ok {
+		state.ebsAttachments[ebsAttachment.AttachmentProperties[apiebs.VolumeIdName]] = ebsAttachment
+	} else {
+		seelog.Debugf("Duplicate ebs attachment information: %v", ebsAttachment)
+	}
+}
+
+// RemoveEBSAttachment removes the ebs volume from state and stops managing
+func (state *DockerTaskEngineState) RemoveEBSAttachment(volumeId string) {
+	if volumeId == "" {
+		seelog.Debug("Cannot remove empty ebs attachment information")
+		return
+	}
+	state.lock.Lock()
+	defer state.lock.Unlock()
+	if _, ok := state.ebsAttachments[volumeId]; ok {
+		delete(state.ebsAttachments, volumeId)
+	} else {
+		seelog.Debugf("Delete non-existed ebs attachment: %v", volumeId)
```

Comment:
nit: 
`seelog.Debugf("RemoveEBSAttachment: The requested EBS attachment: %v does not exist", volumeId)`?

Reply:
added, thanks